### PR TITLE
refactor: moved modules/loss dir into modules/rlhf/loss and updated references

### PR DESCRIPTION
**Pull request by ChinoUkaegbu**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [x] update tests and/or documentation
- [ ] other (please add here)

Addresses issue #1299

#### Changelog
Moved loss directory to rlhf and updated documentation as well as the recipes, configs and tests that imported the loss.

#### Test plan

- [ ] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](../tests/torchtune) for any new functionality
- [ ] update [docstrings](../docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

### CODE REVIEW

File: torchtune/modules/rlhf/loss/dpo.py [hunk 1]
```diff
@@ -255,7 +255,7 @@ class SimPOLoss(nn.Module):
 
     SimPO is pretty much identitcal to DPO but uses average logprobs to eliminate the need for a reference model to regularize
     the policy during training. It also uses a target reward margin to guide the policy towards better responses.
-    This is kind of the same intuition in:class:`~torchtune.modules.loss.IPO`, but instead of optimizing against a margin
+    This is kind of the same intuition in:class:`~torchtune.modules.rlhf.loss.IPO`, but instead of optimizing against a margin
```
Comment (SalmanMohammadi) [inline torchtune/modules/rlhf/loss/dpo.py:258]: small nit;
```suggestion
    This is kind of the same intuition in:class:`~torchtune.modules.rlhf.loss.IPOLoss`, but instead of optimizing against a margin
```

File: docs/source/api_ref_modules.rst [hunk 2]
```diff
@@ -82,11 +82,11 @@ Loss
    :toctree: generated/
    :nosignatures:
 
-   loss.PPOLoss
```
Comment (SalmanMohammadi) [inline docs/source/api_ref_modules.rst:85]: thoughts on moving these as a subsection under RLHF? @joecummings 
not married to either

File: recipes/lora_dpo_single_device.py [hunk 3]
```diff
@@ -56,10 +56,10 @@ class LoRADPORecipeSingleDevice(FTRecipeInterface):
 
 
     The following losses are supported in this recipe:
-        - :class:`~torchtune.modules.loss.DPOLoss`: Direct Preference Optimization (DPO).
-        - :class:`~torchtune.modules.loss.RSOPLoss`: Rejection Sampling Optimization (RSO).
-        - :class:`~torchtune.modules.loss.IPO`: Identity Preference Optimization (IPO).
-        - :class:`~torchtune.modules.loss.SimPOLoss`: Simple Preference Optimization (SimPO).
+        - :class:`~torchtune.modules.rlhf.loss.DPOLoss`: Direct Preference Optimization (DPO).
+        - :class:`~torchtune.modules.rlhf.loss.RSOPLoss`: Rejection Sampling Optimization (RSO).
+        - :class:`~torchtune.modules.rlhf.loss.IPO`: Identity Preference Optimization (IPO).
```
Comment (SalmanMohammadi) [inline recipes/lora_dpo_single_device.py:61]: could you please change
```suggestion
        - :class:`~torchtune.modules.rlhf.loss.IPO`: Identity Preference Optimization (IPO).
```
while you're here? : )

**Comment by ChinoUkaegbu** [comment]
> This looks great, thanks @ChinoUkaegbu!
> 
> Couple things:
> 
> 1. Can you sign the CLA?
> 2. Can you actually move the tests/modules/loss to tests/modules/rlhf/loss as well?

1. Just signed the CLA! I believe it'll take around an hour to reflect
2. Sure, I'll create another branch and push that along with the reviews added.

File: docs/source/api_ref_modules.rst (hunk 2 above)
Comment (joecummings) [reply docs/source/api_ref_modules.rst:85]: Subsection makes sense to me

Comment (SalmanMohammadi) [reply docs/source/api_ref_modules.rst:85]: @ChinoUkaegbu thanks for updating these!
Could you please move the Loss section to under the RLHF section at the bottom of the file?
You'll also need to change the heading from 
```
Loss
------------------
```
to 
```
Losses
^^^^^
```

See [here](https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#sections) for more details

**Comment by ChinoUkaegbu** [comment]
Just incorporated the changes from pre-commit. I'm working on a Windows system which I suspect is why pre-commit was so tedious to setup (I tried installing on WSL and it worked smoothly but when I tried to commit, it reformatted all the files which was weird). Nevertheless, everything should be linted now and once all the checks are complete I'll make an additional commit to move the Loss section to under the RLHF section in the docs :)

File: docs/source/api_ref_modules.rst (hunk 2 above)
Comment (ChinoUkaegbu) [reply docs/source/api_ref_modules.rst:85]: Done!

**Review by joecummings — APPROVED** [review]
Looks great - thanks for the quick fix!!
